### PR TITLE
Fix for ReturnYouTubeDislike freezing on video load.  Logging function improvements

### PR DESCRIPTION
**Pull request by LisousEinaiKyrios**
RYD occasionally freezes the UI when loading a video.  The problem was the UI thread blocks without a timeout.

But I found some other oversights, such as multiple RYD registration calls could be running concurrently (fixed by adding synchronization on the registration call).  Also found the values return by multiple network calls could be reordered, and thus the UI could display the wrong number of dislikes (unlikely, but still possible).  Also, as per RYD api specifications, added the 429 status response (client should backoff on making calls for a brief amount of time). 

Logging now supports lambda functions, which gives free performance since the log messages are not created unless logging is actually enabled.  It also supports automatically detecting which class is making the log call, so no need to pass in a class file anymore (this is done by inspecting the lambda class itself, with no overhead of using reflections).

I will try using my branch for a few days to verify it's working correctly.  But as of now it appears everything is fine.

### CODE REVIEW

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
So far so good, can you seperate the changes to logging to another branch/pr, otherwise squashing the PR will not work with a single commit.

File: app/src/main/java/app/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike.java [hunk 1]
```diff
@@ -36,56 +43,100 @@ public enum Vote {
...
+     */
+    public static synchronized Integer getDislikeCount() {
+        return dislikeCount;
+    }
+    /**
+     * @return true if the videoId parameter matches the current dislike request, and the set value was successful
+     *         If videoID parameter does not match currentVideoId, then this call does nothing
+     */
+    private static synchronized boolean setCurrentDislikeCount(String videoId, Integer videoIdDislikeCount) {
+        if (! videoId.equals(currentVideoId)) {
```
[the remark this answers is not in the document]
Comment (LisousEinaiKyrios) [reply app/src/main/java/app/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike.java:88]: is the Android Studio autoformater setup?  When I run it, nearly ever line gets changed a little.

**Comment by LisousEinaiKyrios** [comment]
> So far so good, can you seperate the changes to logging to another branch/pr, otherwise squashing the PR will not work with a single commit.

The existing logging functions still work, so it should work with all other branches and their changes.

Should I still move the logging changes to a different branch?

**Comment by oSumAtrIX** [comment]
I will push some changes, after that a new PR should be made. This PR should either look into RYD or improving logging so we can properly squash as one commit.

**Comment by LisousEinaiKyrios** [comment]
Ok the logging changes are reverted.  I will make the changes to another branch and do another PR for that.

**Comment by oSumAtrIX** [comment]
I made some but not all comments more consistent and changed a bit of the code. I had to force push because you made some changes after I said to wait for the new PR. If possible, please fix remaining apparent left over inconsistencies.

**Review by LisousEinaiKyrios — COMMENTED** [review]
looks good

**Pull request closed without merging**